Stops a memmove() of a negative amount of memory. It sometimes occurs that
sh_audio->a_out_buffer_len=0, or at least less than playsize, after a seek
has been performed. Not a clean solution, but temporarily stops a segfault
in the code. Causes momentary audible static noise when problem occurs.
--- a/mplayer.c	2007-06-22 20:49:08.000000000 +0800
+++ b/mplayer.c	2007-06-22 20:50:42.000000000 +0800
@@ -1920,6 +1920,8 @@
 	playsize = mpctx->audio_out->play(sh_audio->a_out_buffer, playsize, playflags);
 
 	if (playsize > 0) {
+        if (sh_audio->a_out_buffer_len < playsize)
+            playsize = sh_audio->a_out_buffer_len;
 	    sh_audio->a_out_buffer_len -= playsize;
 	    memmove(sh_audio->a_out_buffer, &sh_audio->a_out_buffer[playsize],
 		    sh_audio->a_out_buffer_len);
